### DISCUSSION TITLE
The names of the via/end-point in the Android notification

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#237] [Q&A] [answered]
Another (probably unimportant for most) request:
Would it be possible to enter the names of the points in the notification, followed by the trip data?

I use Tasker to read this notification to inform interested friends of my distance and arrival time. It is not always clear what a via point is and which is a destination point in my journey. 

Since the data is separated by commas (in K2 there were big dots - probably more readable) it would be good if the name at the beginning of the line was separated in the same way. Of course, if the implementation is not too complicated!

**Comment by devemux86** [comment]
@Ciutnik do you mean the "Settings | Navigation | Advanced | Notifications"?
Have you enabled it?

It contains the next turn instruction with distance, remaining and arrival time for the destination and next via point.
If the next instruction is via point, then the via point name is included.

Or do you mean the line with the next via point distance and time to also include the via point name all the time?

**Reply by Ciutnik** [reply]
Yes, I mean the lines below the turn instructions. Of course, this is for the expanded notification:

![Screenshot_20230707_083918_Cruiser](https://github.com/devemux86/cruiser/assets/122647782/10054a17-bc34-4b40-aef7-5eccf42864d3)

**Comment by devemux86** [comment]
This is easy. @Ciutnik do you mean something like this?

<img src="https://github.com/devemux86/cruiser/assets/3484020/16f9c0f6-c18c-404d-b7ff-a2a457bf8ae5" width="50%"/>

**Comment by devemux86** [comment]
I could also change the separator to dot.
Not sure if it is then easily parsed by external tools analyzing the notification?

**Reply by devemux86** [reply]
Also we probably do not need again the turn instruction "Turn slight left" in the info below?
(as it is already exists in the title)

**Comment by Ciutnik** [comment]
> This is easy... do you mean something like this?

Fantastic! That's what I meant! 👍 Once again a huge thank you!

Tasker has no problem recognizing a large dot as a separator (it did it well in K2), and has the advantage over commas that a comma can be used as a decimal separator. But not everyone uses Tasker - maybe a semicolon would be best as a well-recognized separator in databases? This is probably the most universal symbol, not colliding with other symbols.

As for the repeated turn instruction, it actually seems unnecessary.

Thanks again!

Edit: Maybe a `|` character as a separator?

**Reply by devemux86** [reply]
We could use the dot, as the semicolon is a character that can exist in waypoint names?

**Reply by Ciutnik** [reply]
You're right. If the dot causes a problem in the future, then we'll worry 😉

**Comment by devemux86** [comment, marked as answer]
Waypoint names are available in navigation notifications in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by Ciutnik** [reply]
Thank you very much! Now I need to restore the old version of the notification parsing Task used with K2 🙂
